### ISSUE TITLE
Drop support for Node.js 0.x

### ISSUE THREAD

**Issue by paazmaya** [cst/cst#149]
http://node.green/

While working on the #140 ESLint migration and setting the `google` preset, there seems to be the requirement to use `const` and `let`, as an example for ES6 features that are not available in 0.x versions (without `--harmony`) flag.

How about defining the minimum supported version as `4.2.0` (LTS) and go forward with that?

https://github.com/nodejs/LTS#lts-schedule

**Comment by markelog** [comment]
Yeah, we sure need to do it but for that we would need to bump major version

**Comment by hzoo** [comment]
We can do that now since it's necessary tied to jscs which is deprecated and eslint has already dropped 0.10. It's 0.x in semver so it doesn't have to be major since every update has to be explicitly updated?

**Comment by hzoo** [comment]
Ok we should publish a new version and use node 4 and drop babel-runtime from the dependencies

**Comment by markelog** [comment]
Semver allows it, but it seems lots of people might be upset if we do this. 
We could bump to 1.x so jscs and other conservative consumers wouldn't catch it

**Comment by hzoo** [comment]
It's still an explicit update (no one will automatically get it) so it's ok but if you want we can do 1.x

**Comment by hzoo** [comment]
nvm I see it needs to be a minor after looking at https://semver.npmjs.com/
